src/v2/models/WidgetState.js

```diff
@@ -0,0 +1,128 @@
+/*!
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+define([
+  'okta'
+],
+function (Okta) {
+
+  // Keep track of stateMachine with this special model. Similar to Appstate.js
+  var _ = Okta._;
+
+  return Okta.Model.extend({
+
+    initialize: function (){
+      // add listeners
+      var setRemediationHandler = _.bind(function (transaction) {
+        this.set('remediationSuccess', transaction);
+      }, this);
+      var remediationErrorHandler = _.bind(function (err) {
+        this.set('remediationFailure', err);
+      }, this);
+
+      // Events to set the remediation attributes on the widget state.
+      this.listenTo(this, 'setRemediationSuccess', setRemediationHandler);
```

Comment:
looks like trigger `setRemediationSuccess` is isomorphsim to `state.set('remediationSuccess', resp)`.
curious why still need such indirect invoke?

Reply:
yeah, this got ported over with existing v1 pattern. I could just trigger an event for `remediationSuccess` from the BaseModel and have a listener for it in `WidgetState.js`

Reply:
I guess we dont have to blindly copy the code but refactor along way if v1 implementation doesn't make lot sense now.